src/main/java/net/dv8tion/jda/api/entities/Member.java

```diff
@@ -114,6 +116,26 @@
     @Nullable
     OffsetDateTime getTimeBoosted();
 
+    /**
+     * The time this Member will be released from time out.
+     * <br>If this Member is not in time out, this returns {@code null}.
+     *
+     * @return The time this Member will be released from time out or {@code null} if not in time out
+     */
+    @Nullable
+    OffsetDateTime getTimeUntilTimedOut();
+
+    /**
+     * Whether this Member is in time out.
+     * <br>While a Member is in time out, they cannot reply, send messages, react and speak in voice channels.
+     *
+     * @return True, if this Member is in time out
+     */
+    default boolean isTimedOut()
+    {
+        return getTimeUntilTimedOut() != null;
```

Comment:
This should also check the time if its not null. The event is only received if GUILD_MEMBERS intent is enabled, right?

Reply:
not sure what you mean, but yes, the event is only received with GUILD_MEMBERS

Reply:
I mean that we should check if the timestamp has passed.